internals/cli/masker/matcher.go

```diff
@@ -0,0 +1,91 @@
+package masker
+
+// Matches represents a set of sequence matches. The key is the index at which the match is found and the value is the
+// length of the match. The index corresponds to the index of the byte in the BufferedIndex of the stream.
+type Matches map[int64]int
+
+// Add a new match to the map if it does not yet exist or the existing match has a shorter length.
+func (m Matches) Add(index int64, length int) Matches {
+	existing, exists := m[index]
+	if !exists || existing < length {
+		m[index] = length
+	}
+	return m
+}
+
+// multipleMatcher combines multiple sequenceMatchers to check for matches of secrets against any of them.
+type multipleMatcher struct {
```

Comment:
Does a `multipleMatcher` check for multiple secrets to match against? The name `multiple` is a bit ambiguous. I thought for a second that multiple streams would use the same `multipleMatcher` but that's not the case, right?

Reply:
Renamed after discussing on the phone.